src/Microsoft.AspNet.OData.Shared/Routing/Conventions/RequestMethodExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.OData.Interfaces;
+
+namespace Microsoft.AspNet.OData.Routing.Conventions
+{
+    internal static class RequestMethodExtensions 
+    {
+        internal static ODataRequestMethod GetRequestMethodOrPreflightMethod(this IWebApiRequestMessage request)
+        {
+            if (request.Method == ODataRequestMethod.Options)
+            {
+                IEnumerable<string> values;
+                request.Headers.TryGetValues("Access-Control-Request-Method", out values);
+                var enumerable = values.ToList();
+                if (enumerable.Count()!=0)
+                {
+                    return ToOdataRequestMethod(enumerable.First());
+                }
+
+                return ODataRequestMethod.Unknown;
+            }
+
+            return request.Method;
+        }
+
+        private static ODataRequestMethod ToOdataRequestMethod(string requestMethod)
```

Comment:
@marabooy `ToODataRequestMethod(string requestMethod)` is better for the eyes :)

Reply:
Thanks I chose the last suggestion as it communicates the methods intention clearly